arrow/arrow.py

```diff
@@ -407,12 +412,22 @@ def span_range(cls, frame, start, end, tz=None, limit=None, bounds="[)"):
         """
 
         tzinfo = cls._get_tzinfo(start.tzinfo if tz is None else tz)
-        start = cls.fromdatetime(start, tzinfo).span(frame)[0]
+        start = cls.fromdatetime(start, tzinfo).span(frame, exact=exact)[0]
+        end = cls.fromdatetime(end, tzinfo)
         _range = cls.range(frame, start, end, tz, limit)
-        return (r.span(frame, bounds=bounds) for r in _range)
+        for r in _range:
+            if exact:
+                floor, ceil = r.span(frame, bounds=bounds, exact=exact)
+                if ceil > end:
+                    ceil = end
+                    if bounds[1] == ")":
+                        ceil += relativedelta(microseconds=-1)
+                yield floor, ceil
+            else:
+                yield r.span(frame, bounds=bounds, exact=exact)
```

Comment:
Correct me if I'm wrong here but since `span_range` is using `span` under the hood do we need to add this logic? Seems like we can just pass the `exact` param and be ok.

Reply:
Ah wait I see the reason you've done it like this, ignore the previous comment.